### DISCUSSION TITLE
How to change the binding ports? (specifically the web interface port 80)

### DISCUSSION THREAD

**Discussion by wsw70** [AdguardTeam/AdGuardHome#4608] [Q&A] [answered]
I deployed Adguard Home in a docker environment and I would like to try it as my main DHCP server.

To this, I need to bind it to the host interface but this also means that port 80 would be posted on the host as well. There is already an application using that port.

Is there a way to change the web interface port of Adguard Home to something else?

**Comment by ainar-g** [comment, marked as answer]
We have an [FAQ entry][1] about that.  Please tell us if the FAQ entry is not clear enough.

[1]: https://github.com/AdguardTeam/AdGuardHome/wiki/FAQ#webaddr

**Reply by wsw70** [reply]
Thank you - I just had a look at the main,n page and did not realize there is extensive documentation in the wiki. It works perfect and the FAQ is very clear.

**Comment by ghost** [comment]
Can bind port be the same as login port? Default bind port is 80, but I use TLS encryption certificates for logging into AGH via port 443. I enabled AGH auto redirect feature. My IPTables for Raspberry Pi that runs AGH doesn't allow anyone to connect to Rasperry Pi over port 80, but that doesn't create a problem. If I set binding port to be 443, then I can't login to AGH. I use port 443 on Raspberry Pi only for logging in.

This is more of a network quesiton than AGH issue because I don't fully understand what binding port is and/or what it does.

**Reply by wsw70** [reply]
I am not sure how much I can help you but here you go:

> Can bind port be the same as login port? 

A service that accepts connections does this through an exposed TCP or UDP endpoint which is (roughly speaking) a combination of an IP address and a port. 192.168.110.4:3987 in an example. There can be only one service by port.
When you (as a client) connect to this service, you must 
- know where to connect (= have the full IP/DNS name + port combo)
- know which protocol is expected by the service

There are some services that **traditionally** reside on well-known ports. HTTP is 80, HTTPS is 443, SSH is 22 etc. Thi sis just a convention - there are no technical reasons for that.

Therefore, if you have a service that understands HTTP, it will bind to port 80, and if the service understands HTTPS, it will bind to port 443. Today, you will often have a redirect from port 80° to 443 (i.e. the service directly redirects to HTTPS without attempting to discuss with your client). This is essentially what you stated next:

> Default bind port is 80, but I use TLS encryption certificates for logging into AGH via port 443. I enabled AGH auto redirect feature.

It gest more confusing afterwards:

> My IPTables for Raspberry Pi that runs AGH doesn't allow anyone to connect to Rasperry Pi over port 80, but that doesn't create a problem

So port 80 is not available at all, right? And therefore the redirect is not needed, right? This can be the case, though you would usually want to have 80 exposed with just a redirect.

> If I set binding port to be 443, then I can't login to AGH. I use port 443 on Raspberry Pi only for logging in.

This I don't understand. Wildly guessing - you attempted to bind an HTTP service to a well-known HTTPS port. This won't work if you try to connect via `https://...` to the service because on the other side an HTTP-only service is waiting for you. And won't understand HTTPS.

But you are saying that you use 443 to login, so I am a bit lost here.

**Comment by arboeh** [comment]
I have to response to this:
For me it's impossible to use another web ui port than 80. I tried to change it in AdGuardHome.yaml, in docker-compose.yaml and in the ui setup routine. Nothing works for me. UFW is disabled. Is it hardcoded in the sourcecode?
ADH works great for me, but I just want to use another port, than the common port 80.

**Reply by ainar-g** [reply]
@arboeh, please make sure that you have mounted the volumes correctly and that you are stopping AGH before editing the file.
